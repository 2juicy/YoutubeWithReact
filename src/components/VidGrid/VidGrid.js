import React from "react";
import "./VidGrid.scss";
import VidPreview from "../VidPreview/VidPreview";

export default function VidGrid({ title }) {
  return (
    <>
      <div className="vid-grid-header">
        <span className="title">{title}</span>
      </div>

      <div className="vid-grid">
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
        <VidPreview />
      </div>
    </>
  );
}
