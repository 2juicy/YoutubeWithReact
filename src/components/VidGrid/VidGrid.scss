.vid-grid-header {
  padding: 28px 0 24px 4px;

  &:first-child {
    padding-top: 24px;
  }

  .title {
    font-size: 1.2rem;
    font-weight: bold;
  }
}

.vid-grid {
  display: flex;
  flex-wrap: wrap;
  padding-bottom: 14px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);

  & > * {
    margin: 0 0 24px 4px;
  }
}
