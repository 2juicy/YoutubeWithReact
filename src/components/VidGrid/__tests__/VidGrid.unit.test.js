import React from "react";
import { shallow } from "enzyme";
import VidGrid from "../VidGrid";

describe("VidGrid", () => {
  test("renders without props", () => {
    const wrapper = shallow(<VidGrid />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with title prop", () => {
    const wrapper = shallow(<VidGrid title="Trending" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders without divider", () => {
    const wrapper = shallow(<VidGrid title="Trending" hideDivider />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders with empty string header", () => {
    const wrapper = shallow(<span className="title"></span>);
    expect(wrapper).toMatchSnapshot();
  });
});
