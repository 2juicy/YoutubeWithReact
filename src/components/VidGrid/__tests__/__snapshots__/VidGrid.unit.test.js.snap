// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VidGrid renders with empty string header 1`] = `
<span
  className="title"
/>
`;

exports[`VidGrid renders with title prop 1`] = `""`;

exports[`VidGrid renders without props 1`] = `""`;
